### PR TITLE
Features/customer

### PR DESCRIPTION
**Pull request by NAPATKRUP**

### CODE REVIEW

File: frontend/src/App.tsx [hunk 1]
```diff
@@ -28,6 +29,10 @@ const App: FC = () => {
           <Route exact path="/cart" component={CartPage} />
           <Route exact path="/checkout" component={CheckoutPage} />
           <Route exact path="/payment" component={PaymentPage} />
+          <Route path="/customer" component={CusomterPage} />
+          {/* <Route exact path="/cart/:cartId/payment" component={PaymentPage} /> */}
+          {/* <Route exact path="/manage-product" component={ManageProduct} /> */}
+          {/* <AdminRoute role={user.role} exact path="/manage-product" component={ManageProduct} /> */}
```
Comment (Icyscools) [inline frontend/src/App.tsx]: Remove these lines.

File: frontend/src/components/checkout/page/CheckoutPage.tsx [hunk 2]
```diff
@@ -134,9 +135,7 @@ const CheckoutPage: FC = () => {
           className="border-2 rounded-full bg-dark-100 text-white-100 hover:bg-dark-200 mx-2 p-1"
           onChange={handleAddressIdChange}
         >
-          <option value="" selected>
-            โปรดเลือกที่อยู่ที่ใช้ในการจัดส่ง
-          </option>
+          <option value="">โปรดเลือกที่อยู่ที่ใช้ในการจัดส่ง</option>
```
Comment (Icyscools) [inline frontend/src/components/checkout/page/CheckoutPage.tsx]: Add `disabled` attribute to not letting user select this option.

File: frontend/src/components/customer/page/CustomerOrderDetailPage.tsx [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+  if (orderLoading) {
+    return <Loading />;
+  }
+  if (orderError) {
+    history.push({ pathname: '/error' });
+    return <></>;
+  }
+  const { orderById } = orderData;
+  console.log(orderById);
```
Comment (Icyscools) [inline frontend/src/components/customer/page/CustomerOrderDetailPage.tsx]: Remove console.log

File: frontend/src/components/payment/page/PaymentPage.tsx [hunk 4]
```diff
@@ -65,8 +62,8 @@ const PaymentPage: FC = () => {
   const handleCallBack = (stats: boolean) => {
     if (!stats) {
       toggle();
-      if (isPayment) history.replace({ pathname: '/' });
-      if (isCancel) history.replace({ pathname: '/' });
+      if (isPayment) history.push({ pathname: '/' });
+      if (isCancel) history.push({ pathname: '/' });
```
Comment (Icyscools) [inline frontend/src/components/payment/page/PaymentPage.tsx]: Can combine this to `if (isPayment || isCancel) ...`

File: frontend/src/components/payment/page/PaymentPage.tsx [hunk 5]
```diff
@@ -147,9 +147,7 @@ const PaymentPage: FC = () => {
           className="border-2 rounded-full lg:w-3/12 w-8/12 bg-dark-100 text-white-100 hover:bg-dark-200 p-1"
           onChange={handlePaymentIdChange}
         >
-          <option value="" selected>
-            โปรดเลือกช่องทางในการชำระเงิน
-          </option>
+          <option value="">โปรดเลือกช่องทางในการชำระเงิน</option>
```
Comment (Icyscools) [inline frontend/src/components/payment/page/PaymentPage.tsx]: Add `disabled` for not letting user select this option.

**Review by Icyscools — COMMENTED** [review]
Nice work but can be improve more.

File: frontend/src/components/payment/page/PaymentPage.tsx (hunk 4 above)
Comment (NAPATKRUP) [reply frontend/src/components/payment/page/PaymentPage.tsx]: this condition for future change to orderdetail page after payment

**Comment by NAPATKRUP** [comment]
@Icyscools Ok Done

**Pull request merged**